### PR TITLE
Other Attachments Page

### PR DESCRIPTION
**Pull request by dhaselhan**
* Allow null document types
* Add description to documents

### CODE REVIEW

File: portal-frontend/src/app/features/edit-application/other-attachments/other-attachments.component.scss [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+h4 {
+  margin-bottom: rem(8) !important;
+}
+
+.scrollable {
+  overflow-x: auto;
+}
+
+.mat-mdc-table .mdc-data-table__row {
+  height: 75px;
```
Comment (mhuseinov) [inline portal-frontend/src/app/features/edit-application/other-attachments/other-attachments.component.scss:20]: Pixels?

File: services/apps/portal/src/application/application-document/application-document.controller.ts [hunk 2]
```diff
@@ -78,6 +80,17 @@ export class ApplicationDocumentController {
...
 
+  @Patch('/application/:fileNumber')
+  async update(
+    @Param('fileNumber') fileNumber: string,
+    @Req() req,
+    @Body() body: ApplicationDocumentUpdateDto[],
+  ) {
+    await this.applicationService.verifyAccess(fileNumber, req.user.entity);
+    await this.applicationDocumentService.update(body, fileNumber);
+    return;
```
Comment (mhuseinov) [inline services/apps/portal/src/application/application-document/application-document.controller.ts:91]: return nothing?

File: services/apps/portal/src/application/application.service.ts [hunk 3]
```diff
@@ -202,7 +202,7 @@ export class ApplicationService {
           typeCode: application.typeCode,
           dateSubmittedToAlc: Date.now().toString(),
           documents: application?.documents.map((d) => ({
-            type: d.type,
+            type: d.type!, //TODO: Do we verify this?
```
Comment (mhuseinov) [inline services/apps/portal/src/application/application.service.ts:205]: I think we do check file type on api level

File: services/apps/portal/src/common/automapper/application-owner.automapper.profile.ts [hunk 4]
```diff
@@ -36,6 +36,7 @@ export class ApplicationOwnerProfile extends AutomapperProfile {
                 uuid: a.corporateSummary.uuid,
                 fileName: a.corporateSummary.fileName,
                 type: DOCUMENT_TYPE.CORPORATE_SUMMARY,
+                description: null,
```
Comment (mhuseinov) [inline services/apps/portal/src/common/automapper/application-owner.automapper.profile.ts:39]: is this null intentional?

File: services/apps/portal/src/application/application.service.ts (hunk 3 above)
Comment (dhaselhan) [reply services/apps/portal/src/application/application.service.ts:205]: This idea is around checking that each document has a type before we submit to ALCS, we might need some other validation steps as well as we start building out more features.

File: services/apps/portal/src/common/automapper/application-owner.automapper.profile.ts (hunk 4 above)
Comment (dhaselhan) [reply services/apps/portal/src/common/automapper/application-owner.automapper.profile.ts:39]: yes, corporateSummaries cannot have descriptions

File: services/apps/portal/src/application/application-document/application-document.controller.ts (hunk 2 above)
Comment (dhaselhan) [reply services/apps/portal/src/application/application-document/application-document.controller.ts:91]: What should I return?

Comment (mhuseinov) [reply services/apps/portal/src/application/application-document/application-document.controller.ts:91]: you could do something like return this.applicationDocumentService.update(body, fileNumber). Where update on service side does return repo.update(array)

File: portal-frontend/src/app/features/edit-application/other-attachments/other-attachments.component.scss (hunk 1 above)
Comment (dhaselhan) [reply portal-frontend/src/app/features/edit-application/other-attachments/other-attachments.component.scss]: Fixed

File: services/apps/portal/src/application/application-document/application-document.controller.ts (hunk 2 above)
Comment (dhaselhan) [reply services/apps/portal/src/application/application-document/application-document.controller.ts]: Done

**Review by mhuseinov — APPROVED** [review]
🎯

**Pull request merged**